pkg/kstatus/status/status_compute_test.go

```diff
@@ -1269,6 +1269,85 @@ func TestJobStatus(t *testing.T) {
 	}
 }
 
+var hpaNoStatus = `
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: test
+  namespace: qual
+  generation: 1
```

Comment:
Does HPA have an observedGeneration?

Usually we expect that observedGeneration must equal generation, otherwise the object is still reconciling.

Ideally the CRD would require observedGeneration and default it to zero, that way the status is always defined after being applied, even if the controller hasn't updated the status yet.